sympy/ntheory/generate.py

```diff
@@ -219,10 +242,31 @@ def prime(nth):
         primerange : Generate all primes in a given range
         primepi : Return the number of primes less than or equal to n
     """
-    n = as_int(nth)
+    n = int(nth)
```

Comment:
`as_int` is used to guarantee that the input is an integer and to raise an error if it's not. I guess we should add a test for something like `prime(3.14)`